integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/message/FileMessagePersister.java

```diff
@@ -26,44 +26,78 @@ public FileMessagePersister(IntegrasjonspunktProperties props) {
+        File localFile = new File(filedir + filename);
         localFile.getParentFile().mkdirs();
 
-        if (props.getNextmove().getApplyZipHeaderPatch() && props.getNextmove().getAsicfile().equals(filename)){
-            BugFix610.applyPatch(message, cr.getConversationId());
+        if (props.getNextmove().getApplyZipHeaderPatch() && ASIC_FILE.equals(filename)) {
+            BugFix610.applyPatch(message, messageId);
         }
 
         try (FileOutputStream os = new FileOutputStream(localFile);
              BufferedOutputStream bos = new BufferedOutputStream(os)) {
             bos.write(message);
-            bos.flush();
-            bos.close();
         } catch (IOException e) {
-            log.error("Could not write asic container to disc.", e);
+            log.error("Could not write asic container to disk.", e);
             throw e;
         }
     }
 
     @Override
-    public byte[] read(ConversationResource cr, String filename) throws IOException {
-        String filedir = getConversationFiledirPath(cr);
-        File file = new File(filedir+filename);
+    public void writeStream(String messageId, String filename, InputStream inputStream, long size) throws IOException {
+        String filedir = getMessageFiledirPath(messageId);
+        File localFile = new File(filedir + filename);
+        localFile.getParentFile().mkdirs();
+
+        try (FileOutputStream os = new FileOutputStream(localFile);
+             BufferedOutputStream bos = new BufferedOutputStream(os)) {
+            int bytes = IOUtils.copy(inputStream, bos);
+            log.debug("Storing {}: {}", filename, humanReadableByteCount(bytes));
+        } catch (IOException e) {
+            log.error("Could not write asic container to disk.", e);
+            throw e;
+        }
+    }
+
+    private String humanReadableByteCount(int bytes) {
+        int unit = 1000;
+        if (bytes < unit) return bytes + " B";
+        int exp = (int) (Math.log(bytes) / Math.log(unit));
+        String pre = "kMGTPE".charAt(exp - 1) + "";
+        return String.format("%.1f %sB", bytes / Math.pow(unit, exp), pre);
+    }
+
+    @Override
+    public byte[] read(String messageId, String filename) throws IOException {
+        String filedir = getMessageFiledirPath(messageId);
+        File file = new File(filedir + filename);
         return FileUtils.readFileToByteArray(file);
     }
 
     @Override
-    public void delete(ConversationResource cr) throws IOException {
-        File dir = new File(getConversationFiledirPath(cr));
+    public FileEntryStream readStream(String messageId, String filename) {
+        String filedir = getMessageFiledirPath(messageId);
+        File file = new File(filedir + filename);
+        try {
+            BufferedInputStream fis = new BufferedInputStream(new FileInputStream(file));
+            return FileEntryStream.of(fis, file.length());
+        } catch (FileNotFoundException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)